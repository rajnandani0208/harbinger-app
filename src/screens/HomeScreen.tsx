// screens/HomeScreen.tsx
import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { ScrollView, Text, TouchableOpacity, View } from 'react-native';
import styles from './HomeScreenStyle';
const HomeScreen: React.FC = () => {
    const navigation = useNavigation();

    const navigateToPoll = () => {
        navigation.navigate('PollList'); // Navigate to Poll screen
    };
    const navigateToForm = () => {
        navigation.navigate('FormScreen'); // Navigate to Form screen
    };
    const navigateToFormList = () => {
        navigation.navigate('FormList'); // Navigate to Form screen
    };
    const navigateToPollList = () => {
        navigation.navigate('PollSummary', { pollId: 1 })
    };
    return (
        <View style={styles.container}>
            <ScrollView contentContainerStyle={styles.scrollStyle} showsVerticalScrollIndicator={false} >
                <Text style={styles.title}>Welcome to the App</Text>
                <Text style={styles.subtitle}>Choose an option to get started:</Text>
                <TouchableOpacity style={styles.card} onPress={navigateToForm}>
                    <Text style={styles.cardTitle}>Form</Text>
                    <Text style={styles.cardDescription}>Fill out forms and provide your feedback.</Text>
                </TouchableOpacity>


                <TouchableOpacity style={styles.card} onPress={navigateToFormList}>
                    <Text style={styles.cardTitle}>Form List</Text>
                    <Text style={styles.cardDescription}>View Submitted Forms</Text>
                </TouchableOpacity>


                <View style={styles.cardContainer}>
                    <TouchableOpacity style={styles.card} onPress={navigateToPoll}>
                        <Text style={styles.cardTitle}>Poll</Text>
                        <Text style={styles.cardDescription}>Participate in polls and share your opinion.</Text>
                    </TouchableOpacity>

                    <TouchableOpacity style={styles.card} onPress={navigateToPollList}>
                        <Text style={styles.cardTitle}>Poll summary</Text>
                        <Text style={styles.cardDescription}>View Poll Summary</Text>
                    </TouchableOpacity>


                </View>
            </ScrollView>
        </View>
    );
};



export default HomeScreen;