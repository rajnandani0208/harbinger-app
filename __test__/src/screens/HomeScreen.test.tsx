import React from 'react';
import { fireEvent, render } from '@testing-library/react-native'
import HomeScreen from '@/src/screens/HomeScreen';
import { NavigationContainer } from '@react-navigation/native';

const renderWithNavigation = (ui: React.ReactNode, options = {}) => {
    return render(<NavigationContainer>{ui}</NavigationContainer>, options);
};

// Mock the navigation object
const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
    return {
        ...jest.requireActual('@react-navigation/native'),
        useNavigation: () => ({
            navigate: mockNavigate,
        }),
    };
});

describe('<HomeScreen /> rendering', () => {
    it('Home screen renders correctly', () => {
        const screen = renderWithNavigation(<HomeScreen />);
        const tree = screen.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles FormScreen navigation correctly', async () => {
        const { findByTestId } = renderWithNavigation(<HomeScreen />);
        const gotoFormBtn = await findByTestId('gotoFormBtn');

        fireEvent.press(gotoFormBtn);
        expect(mockNavigate).toHaveBeenCalledWith('FormScreen');
    })

    it('handles FormList navigation correctly', async () => {
        const { findByTestId } = renderWithNavigation(<HomeScreen />);
        const gotoFormListBtn = await findByTestId('gotoFormListBtn');

        fireEvent.press(gotoFormListBtn);
        expect(mockNavigate).toHaveBeenCalledWith('FormList');
    })

    it('handles PollList navigation correctly', async () => {
        const { findByTestId } = renderWithNavigation(<HomeScreen />);
        const gotoPollBtn = await findByTestId('gotoPollBtn');

        fireEvent.press(gotoPollBtn);
        expect(mockNavigate).toHaveBeenCalledWith('PollList');
    })

    it('handles PollSummary navigation correctly', async () => {
        const { findByTestId } = renderWithNavigation(<HomeScreen />);
        const gotoPollSummary = await findByTestId('gotoPollSummary');

        fireEvent.press(gotoPollSummary);
        expect(mockNavigate).toHaveBeenCalledWith('PollSummary', { pollId: 1 });
    })

});