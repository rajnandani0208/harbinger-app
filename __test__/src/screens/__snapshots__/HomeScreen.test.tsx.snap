// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> rendering Home screen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f9f9f9",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Text
        style={
          {
            "fontSize": 28,
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        Welcome to the App
      </Text>
      <Text
        style={
          {
            "fontSize": 18,
            "marginBottom": 30,
            "textAlign": "center",
          }
        }
      >
        Choose an option to get started:
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#4682B4",
            "borderRadius": 10,
            "elevation": 3,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 4,
            "width": "90%",
          }
        }
        testID="gotoFormBtn"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          Form
        </Text>
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 16,
            }
          }
        >
          Fill out forms and provide your feedback.
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#4682B4",
            "borderRadius": 10,
            "elevation": 3,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 4,
            "width": "90%",
          }
        }
        testID="gotoFormListBtn"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          Form List
        </Text>
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 16,
            }
          }
        >
          View Submitted Forms
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#4682B4",
              "borderRadius": 10,
              "elevation": 3,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 20,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": "90%",
            }
          }
          testID="gotoPollBtn"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Poll
          </Text>
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 16,
              }
            }
          >
            Participate in polls and share your opinion.
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#4682B4",
              "borderRadius": 10,
              "elevation": 3,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 20,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 4,
              "width": "90%",
            }
          }
          testID="gotoPollSummary"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Poll summary
          </Text>
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 16,
              }
            }
          >
            View Poll Summary
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
